src/crypto/sha1.cpp

```diff
@@ -151,6 +151,15 @@ CSHA1::CSHA1() : bytes(0)
     sha1::Initialize(s);
 }
 
+CSHA1::CSHA1(const unsigned char iv[OUTPUT_SIZE]) : bytes(0)
+{
+    s[0] = ReadBE32(iv);
+    s[1] = ReadBE32(iv + 4);
+    s[2] = ReadBE32(iv + 8);
+    s[3] = ReadBE32(iv + 12);
+    s[4] = ReadBE32(iv + 16);
```

Comment:
I'm being consistent with the defined serializations for the algorithm, right?

There are two reasons: pragmatic code reuse and standard compliance. `Midstate` and `Finalize` share code, and the serialization of the latter is defined both by consensus and official standard to be big endian for all except RIPEMD.

Midstate is technically not defined by any standard, but by being consistent with the serialization which is defined for the final value this has the nice property that it interfaces well with other implementations. You should be able to drop the SHA-256 midstate into any other implementation and it will just work Likewise if you manually do the padding and then extract midstate, you'll end up bit-for-bit with the same value as if you had done a normal hash.

Principle of least surprise.